blockchain/pallets/minting/src/lib.rs

```diff
@@ -152,13 +152,13 @@ pub mod pallet {
         #[pallet::weight((
             10_000 + T::DbWeight::get().reads_writes(2, 2),
             DispatchClass::Normal,
-            Pays::No
+            Pays::Yes
```

Comment:
I don't think the Substrate node will accept this transaction if we have this as Pays::Yes.

Reply:
Interesting, just checking the returns from this pallet test does not ensure it will be accepted by the node, the test runtime differs as it the test runtime does not go through the  System pallet and the Transaction Payment pallet. Makes sense, as such Pays::No also excludes transaction costs of the intrinsic. I was not aware of these details.

Reply:
Does Substrate accept the transaction if the static is Pays::Yes? I don't think it would for an account with no balance. I would expect it to work in a unit test, but not when running a local node.

It should be fine to statically specify Pays::No and override with Pays::Yes for the relevant cases.